tests/rbd_mirror/rbd_mirror_reconfigure_primary_cluster.py

```diff
@@ -0,0 +1,101 @@
+    1) Perform the mirroring bootstrap for cluster peers
+    2) copy and import bootstrap token to peer cluster
+    3) verify peer cluster got added successfully after failback
+    4) verify all the images from secondary mirrored to primary
+    5) Demote initially promoted secondary as primary
+    6) Promote newly created mirrored cluster as primary
+"""
+import time
+
+from tests.rbd.rbd_utils import Rbd
+from tests.rbd_mirror import rbd_mirror_utils as rbdmirror
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(**kw):
+    """Verification of rbd mirroring failback scenarios on primary site,
+    This module configures the mirroring back to failback cluster.
+
+    Args:
+        kw: test data
+        Example::
+            config:
+                poolname: pool_failback
+    Returns:
+        int: The return value - 0 for success, 1 for failure
+    """
+    try:
+        log.info("Re-establishing mirror with newly created cluster as Primary")
+        config = kw.get("config")
+        poolname = kw.get("poolname")
+
+        mirror1 = rbdmirror.RbdMirror(
+            kw.get("ceph_cluster_dict").get("ceph-rbd3"), config
+        )
+
+        mirror2 = rbdmirror.RbdMirror(
+            kw.get("ceph_cluster_dict").get("ceph-rbd2"), config
+        )
+
+        rbd1, rbd2, rbd3 = [
+            Rbd(**kw, req_cname=cluster_name)
+            for cluster_name in kw["ceph_cluster_dict"].keys()
+        ]
+
+        mirror1.create_pool(poolname=poolname)
+        mirror1.config_mirror(mirror2, poolname=poolname, mode="image")
+        mirror1.wait_for_status(poolname=poolname, health_pattern="OK")
+
+        mirror1_image_list = rbd3.list_images(poolname)
+        mirror2_image_list = rbd2.list_images(poolname)
+
+        if mirror1_image_list.sort() == mirror2_image_list.sort():
+            log.info(
+                "After failback, secondary cluster data successfuly migrated to primary cluster"
+            )
+        else:
+            log.error(
+                "After failback, secondary cluster data are not successfuly migrated to primary cluster"
+            )
+            return 1
+
+        # To run I/O on only first image
+        is_first_image = True
+        for imagename in mirror1_image_list:
+            imagespec = poolname + "/" + imagename
+            if is_first_image:
+                mirror2.benchwrite(imagespec=imagespec, io=config.get("io-total"))
+                is_first_image = False
+                time.sleep(60)
+                mirror2.check_data(peercluster=mirror1, imagespec=imagespec)
+
+            # Demote initially promoted secondary as primary
+            mirror2.demote(imagespec=imagespec)
+            mirror2.wait_for_status(imagespec=imagespec, state_pattern="up+stopped")
+
+            # Promote newly created mirrored cluster as primary
+            mirror1.promote(imagespec=imagespec)
+            mirror1.wait_for_status(imagespec=imagespec, state_pattern="up+stopped")
```

Comment:
we should also add check for secondary cluster image status, once we do promote on primary cluster. IMO image status should be "up+replaying".

Reply:
Done